stm/vernac_classifier.ml

```diff
@@ -99,15 +98,18 @@ let classify_vernac e =
       VtStartProof(Doesn'tGuaranteeOpacity, idents_of_name i), VtLater
 
     | VernacDefinition (_,({v=i},_),ProveBody _) ->
-       let guarantee = if poly then Doesn'tGuaranteeOpacity else GuaranteesOpacity in
-        VtStartProof(guarantee, idents_of_name i), VtLater
+      let polymorphic = Attributes.(parse_drop_extra polymorphic atts) in
+      let guarantee = if polymorphic then Doesn'tGuaranteeOpacity else GuaranteesOpacity in
```

Comment:
Maybe this code pattern should be lifted to some function `let guarantee_if_mono atts = if Attributes.(parse_drop_extra polymorphic atts) then Doesn'tGuaranteeOpacity else GuaranteesOpacity`

Reply:
This was already there and its removal is being studied in another PR, so I'll leave it as it is for now.